packages/updating-element/src/updating-element.ts

```diff
@@ -534,6 +537,26 @@ export abstract class UpdatingElement extends HTMLElement {
    */
   private _reflectingProperty: PropertyKey | null = null;
 
+  /**
+   * Set of callbacks called in connectedCallback.
+   */
+  connectedCallbacks: Set<connectCallback> = new Set();
+
+  /**
+   * Set of callbacks called in disconnectedCallback.
+   */
+  disconnectedCallbacks: Set<connectCallback> = new Set();
```

Comment:
Did you explore lazily creating the collections? Then at callback time you could do `this.connectedCallbacks?.forEach(...)`

Reply:
The set is lazily created now.